src/cursor.rs

```diff
@@ -125,7 +124,7 @@ impl FreeList {
         for _ in 0..FREE_LIST_LEN {
             res.try_push(&mut Rc::new(NodeData {
                 kind: Kind::Free { next_free: None },
-                green: ptr::NonNull::dangling(),
```

Comment:
Hm, `ptr::NonNull::danging` seems much more straightforward to me than the dummy node shenanigans. Why do we need this change?

Reply:
`ptr::NonNull::dangling` (as well as `ptr::dangling` requires `T: Sized`).

The next PR changes to provide a `dangling` fn.